File: CS/WinSample.Module/Module.cs
using System;
using System.Collections.Generic;

using DevExpress.ExpressApp;
using System.Reflection;
using DevExpress.ExpressApp.Model;


namespace WinSample.Module {
    public interface IModelListViewExt : IModelNode {
        string AdditionalCriteria { get; set; }
    }

    public sealed partial class WinSampleModule : ModuleBase {
        public WinSampleModule() {
            InitializeComponent();
        }
        public override void ExtendModelInterfaces(ModelInterfaceExtenders extenders) {
            base.ExtendModelInterfaces(extenders);
            extenders.Add<IModelListView, IModelListViewExt>();
        }
    }
}

File: VB/WinSample.Module/Module.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic

Imports DevExpress.ExpressApp
Imports System.Reflection
Imports DevExpress.ExpressApp.Model


Namespace WinSample.Module
	Public Interface IModelListViewExt
	Inherits IModelNode
		Property AdditionalCriteria() As String
	End Interface

	Public NotInheritable Partial Class WinSampleModule
		Inherits ModuleBase
		Public Sub New()
			InitializeComponent()
		End Sub
		Public Overrides Sub ExtendModelInterfaces(ByVal extenders As ModelInterfaceExtenders)
			MyBase.ExtendModelInterfaces(extenders)
			extenders.Add(Of IModelListView, IModelListViewExt)()
		End Sub
	End Class
End Namespace
